### DISCUSSION TITLE
HLK-2420 Radar Modul

### DISCUSSION THREAD

**Discussion by kodibrain** [arendst/Tasmota#20875] [Ideas]
Hi, is it possible to add the HLK-2420 sensor? Currently, only the HLK-2410 is supported by Tasmota. The 2420 has the advantage that its pins have a 2.54mm spacing and supposedly covers a larger range.
Thanks

**Comment by sfromis** [comment]
If it is compatible when it comes to commands/response, it may be easy. Otherwise, some driver work may be needed.

**Comment by Staars** [comment]
Are you aware that the HLK2410C is basically a 2410B with a spacing of 2.54mm?

https://www.hlktech.net/index.php?id=1095

**Reply by kodibrain** [reply]
Oh, really? I didn't know that. Thanks for the heads up. I have the 2420 left over from another project with esphome. That's why I asked.

**Reply by Staars** [reply]
Okay, this thing is cheap enough to buy one for testing.
I just ordered one and will eventually try my luck with Berry.

This may take awhile…

**Reply by Staars** [reply]
... I started with a generic approach for the LD24xx family in Berry.
https://github.com/Staars/berry-examples/blob/main/serial/serial_ld24xx.be

The LD2420 is the first part of this driver in alpha state. You may give it a try.
